sdk/nodejs/runtime/closure/serializeClosure.ts

```diff
@@ -519,3 +571,54 @@ function envObjToString(envObj: Record<string, string>): string {
         .map((k) => `${k}: ${envObj[k]}`)
         .join(", ")} }`;
 }
+
+/**
+ * An identifier to be imported into a serialised function.
+ */
+interface ImportedIdentifier {
+    /**
+     * True if and only if the identifier to be imported would shadow a reserved
+     * identifier (e.g. "exports").
+     */
+    reserved: boolean;
+
+    /**
+     * The identifier required by serialised function code.
+     */
+    identifier: string;
+
+    /**
+     * An alias for the required identifier that doesn't clash with reserved
+     * identifiers. May be the required identifier itself if it doesn't clash.
+     */
+    as: string;
+}
+
+/**
+ * Computes an appropriate {@link ImportedIdentifier} for a given identifier.
+ */
+function importIdentifier(identifier: string): ImportedIdentifier {
+    if (RESERVED_IDENTIFIERS.has(identifier)) {
+        const as = `${identifier}_${Math.random().toString(36).substring(7)}`;
```

Comment:
This makes it difficult to write a snapshot closure test against this, could we just prefix with some very unlikely prefix like "__pulumi_closure_import_" and then add a test for this case as well?

Reply:
:+1: on using a prefix, In https://github.com/pulumi/pulumi/pull/15887 I picked `__pulumi_...`, so something similar would be good.

For the tests look at `sdk/nodejs/tests/runtime/testdata/closure-tests/cases`. You can run the tests with `make test_closure` inside `sdk/nodejs`.

Reply:
:+1: